app/controllers/landing_page_controller.rb

```diff
@@ -94,23 +83,37 @@ def community_id(request)
     end
   end
 
-  def render_landing_page(structure)
+  def render_landing_page(cid, structure)
+    locale = structure["settings"].values_at("locale")
+
     render :landing_page,
            locals: { font_path: "/landing_page/fonts",
                      styles: landing_page_styles,
                      javascripts: {
                        location_search: location_search_js
                      },
-                     sections: denormalizer.to_tree(structure) }
+                     sections: denormalizer(cid, locale).to_tree(structure) }
   end
 
   def render_not_found(msg = "Not found")
     self.status = 404
     self.response_body = msg
   end
 
-  def append_asset_path(asset)
-    asset.merge("src" => ["landing_page", asset["src"]].join("/"))
+  def marketplace_data(cid, locale)
+    primary_color = Community.where(id: cid)
+                    .pluck(:custom_color1)
+                    .first
+
+    name, slogan, description = CommunityCustomization
+                                .where(community_id: cid, locale: locale)
+                                .pluck(:name, :slogan, :description)
+                                .first
+
+    { "primary_color" => primary_color ? "#" + primary_color : nil,
```

Comment:
What about using `.present?` there to make sure that empty string is threated as falsy

`primary_color.present? ? "#" + primary_color : nil,`